jdaviz/configs/cubeviz/plugins/tools.py

```diff
@@ -42,3 +47,133 @@ def on_mouse_event(self, data):
         self.viewer.session.hub.broadcast(msg)
 
         self._time_last = time.time()
+
+
+@viewer_tool
+class SpectrumPerSpaxel(CheckableTool):
+
+    icon = os.path.join(ICON_DIR, 'pixelspectra.svg')
+    tool_id = 'jdaviz:spectrumperspaxel'
+    action_text = 'See spectrum at a single spaxel'
+    tool_tip = 'Click on the viewer and see the spectrum at that spaxel in the spectrum viewer'
+
+    def __init__(self, viewer, **kwargs):
+        self.old_y_min = 0
+        self.old_y_max = 0
+        self.spectrum_viewer = None
+
+        # Keep track of data created during this session and remove
+        # them from the spectrum viewer after the tool is deactivated
+        self.created_data_labels = []
+
+        # #b515d1 is magenta
+        self.colors = ['#b515d1']
+        super().__init__(viewer, **kwargs)
+
+    def activate(self):
+        self.viewer.add_event_callback(self.on_mouse_event,
+                                       events=['click'])
+        self.spectrum_viewer = self.viewer.jdaviz_app.get_viewer_by_id("cubeviz-3")
+
+    def deactivate(self):
+        self.viewer.remove_event_callback(self.on_mouse_event)
+        self.remove_highlight_on_pixel()
+        for label in self.created_data_labels:
+            self.viewer.jdaviz_app.remove_data_from_viewer("spectrum-viewer", label)
+        self.created_data_labels = []
+
+    def on_mouse_event(self, data):
+        event = data['event']
+
+        if event == 'click':
+            event_x = data['domain']['x']
+            event_y = data['domain']['y']
+
+            x = round(event_x)
+            y = round(event_y)
+
+            self.remove_highlight_on_pixel()
+            self.highlight_pixel(x, y)
+
+            # Check if there are any cubes in the viewer
+            cube_in_viewer = False
+            dc = self.viewer.session.data_collection
+
+            # Spectrum is created for each active layer in the viewer
+            for data in self.viewer.data():
+                if len(data.shape) != 3:
+                    continue
+                cube_in_viewer = True
+
+                cube = data.get_object(cls=Spectrum1D, statistic=None)
+                if x > cube.shape[0] or y > cube.shape[1] or x < 0 or y < 0:
+                    return
+
+                spec = Spectrum1D(flux=cube.flux[x][y], spectral_axis=cube.spectral_axis,
+                                  meta=cube.meta)
+                label = f"{data.label}_at_pixel"
+
+                # Remove data from viewer, re-add data to app, and then add data
+                # back to spectrum viewer
+                self.viewer.jdaviz_app.remove_data_from_viewer("spectrum-viewer", label)
+                dc[label] = spec
```

Comment:
is there any difference between this and calling `app.add_data` (looks like the only thing calling `add_data` would add is some label checks and a snackbar message)?